x-pack/plugins/enterprise_search/jest.config.js

```diff
@@ -8,4 +8,11 @@ module.exports = {
   preset: '@kbn/test',
   rootDir: '../../..',
   roots: ['<rootDir>/x-pack/plugins/enterprise_search'],
+  collectCoverage: true,
```

Comment:
I don't think you want to enable coverage by default. In your jest.sh file you can pass `--coveage` to do this.

Reply:
Hm, from a team perspective I'd prefer it for this project - coverage is fairly important to us and something I want our front end devs to always have in front of them.

If this has implications for CI or affects processes outside just our team/code though I can definitely turn it off and only set it in `jest.sh` - if not, I'd prefer to leave it as the default

Reply:
I appreciate the importance of code coverage. While it won't have an impact on CI it does impact the default experience of anyone running `yarn jest:test` within the plugin. It will by default include code coverage for the entire plugin which requires a lot of scrolling to get to the normal output.

I will note that we are working on having code coverage reports/metrics as part of the pull request details.

Reply:
> While it won't have an impact on CI it does impact the default experience of anyone running yarn jest:test within the plugin

This should mostly be just Enterprise Search frontend devs though, right? In which case our team would fully expect this behavior. Is the concern for other non-Enterprise Search devs who need to make changes to our files/repo? (e.g., Kibana ops, etc.) That's definitely valid if so, but I would argue that it's a minor nuisance at worst for non-Enterprise Search devs who rarely need to make changes / run tests directly on our repo, but it's a major quality of life default for Enterprise Search devs since we're working on our code constantly and essentially always want to see accompanying coverage %s when we're running tests.

> I will note that we are working on having code coverage reports/metrics as part of the pull request details.

Oh wow, that's amazing! I love developing on the Kibana platform so much, y'all just continue to be blow me away 🎉  Just curious, will the code coverage report for the touched files in the PR only, or reported based on the nearest jest config file (i.e. the entire plugin)?

Reply:
> This should mostly be just Enterprise Search frontend devs though, right? In which case our team would fully expect this behavior. Is the concern for other non-Enterprise Search devs who need to make changes to our files/repo? (e.g., Kibana ops, etc.) That's definitely valid if so, but I would argue that it's a minor nuisance at worst for non-Enterprise Search devs who rarely need to make changes / run tests directly on our repo, but it's a major quality of life default for Enterprise Search devs since we're working on our code constantly and essentially always want to see accompanying coverage %s when we're running tests.

Sounds like you have thought it through. I was just raising to make sure you're aware and sounds like this is what you expect. :+1: 

> Oh wow, that's amazing! I love developing on the Kibana platform so much, y'all just continue to be blow me away tada Just curious, will the code coverage report for the touched files in the PR only, or reported based on the nearest jest config file (i.e. the entire plugin)?

It will be for the touched files. I really feel that is where the value of a code coverage report comes in, and it must be available at review time. The first step of this is to include the change as part of the metrics. However, we intend to iterate on this over time.

Reply:
> It will be for the touched files. I really feel that is where the value of a code coverage report comes in, and it must be available at review time.

Love it, that's totally what I was hoping for! Can't wait to see it happen, I'll definitely owe you some beers at the next all hands (whenever that is, haha 🍻 )

Reply:
95% of the time that I'm running jest it's with the `--watch` flag, in which case, I do not need coverage reports, and probably don't want them to be run if it slows down the speed for tests. Do you know what effect this has on that use case?

Reply:
@JasonStoltz With `--watch` on, it reports coverage for only the file that you touched:

<img width="715" alt="" src="https://user-images.githubusercontent.com/549407/102513102-a6c55680-403f-11eb-885b-123db97515a1.png">

That was on a server ts file, when I made changes on a Kea logic file and on a component tsx file, I saw some different/odd behavior (all 0s) but still nothing (IMO) that would significantly affect speed since coverage was still only being reported for 1 file vs the entire plugin.

I'm going to go ahead and merge this plugin in now as a heads up since it's been open a while - if you see significant performance issues due to text coverage reports, definitely let me know and we can continue to adjust.

Reply:
Ah, well that is actually really rad if it only reports coverage on the file you touched. That's a big plus. I was seeing different behavior when I tried it out.

Reply:
I think it's different now with Tyler's changes + (perhaps?) the jest config changes in this PR - definitely feel free to ping if you you see otherwise locally on latest master!